main.tf

```diff
@@ -140,8 +140,9 @@ resource "google_filestore_instance" "rwx" {
   }
 
   networks {
-    network = module.vpc.network_name
-    modes   = ["MODE_IPV4"]
+    network           = module.vpc.network_name
+    modes             = ["MODE_IPV4"]
+    reserved_ip_range = var.filestore_subnet_cidr
```

Comment:
From comment above : Only used when `storage_type="ha" Is there a check here? Should this not be:
```
reserved_ip_range = var.storage_type == "ha" ? var.file_store_subnet_cidr : null
```

Reply:
the guard is already in the resource
  count  = var.storage_type == "ha" ? 1 : 0